integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/column/InputFieldsCollector.java

```diff
@@ -60,13 +68,63 @@ static void collect(
 
   private static void discoverInputsFromNode(
       OpenLineageContext context, LogicalPlan node, ColumnLevelLineageBuilder builder) {
-    extractDatasetIdentifier(context, node).stream()
+    List<DatasetIdentifier> datasetIdentifiers = extractDatasetIdentifier(context, node);
```

Comment:
I would be really in favor of `JdbcColumnLineageCollector` with methods `extractExternalInputs` and `extractExpressionsFromJDBC` with some unit tests attached it.